andela/ah-the-unsullied
# #162949212 Send verification email on register

**What does this PR do?**
- Sends an email to the user with an activation link upon registration

**Description of the Task to be completed**
- generate an account activation link.
- send email on registration.
- activate user account upon click on link
- test verification email and activation functionality.

**How should this be manually tested?**
- Clone the repository: git clone https://github.com/andela/ah-the-unsullied.git
- cd ah-the-unsullied
- git checkout ft-activate-account-162949212
- create a new Gmail account
- [sign in and allow access to less secure apps](https://myaccount.google.com/lesssecureapps)
- create .env on the ah-the-unsullied folder and add the following
```export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/NAME'
     export SECRET_KEY="<your_secret_key_here>"
     export HOST_PASSWORD="<Gmail_Password>"
     export HOST_USER="<Gmail_Username>"
     export SMTP_HOST='smtp.gmail.com'
     export EMAIL_PORT='587
```
- Run the following commands 
```python3 -m venv venv
     source .env
     python manage.py migrate
     python manage.py runserver
```
- open Postman and test the app by registering a user with an existing email.
- check mail inbox for an activation link.
   
**Relevant PT story**
#162949212

**Screenshots**
<img width="873" alt="screenshot 2019-01-21 at 12 51 04" src="https://user-images.githubusercontent.com/11676467/51466819-1fe8a000-1d7c-11e9-812f-467a88397150.png">
